firedrake/preconditioners/pmg.py

```diff
@@ -262,6 +262,35 @@ def prolongation_transfer_kernel_aij(Pk, P1):
     return kernel
 
 
+def tensor_product_space_query(V):
+    # Check if a function space V is a tensor product of
+    # either CG(N) or DG(N) on quads or hexes (same N along every direction).
+    # Additionally, return the degree N and a list of strings with the family.
+    ele = V.ufl_element()
+    cell = ele.cell()
+    isquad = cell.cellname().find("quadrilateral") >= 0  # hex == "quadrilateral * interval"
+
+    if isinstance(ele, (firedrake.VectorElement, firedrake.TensorElement)):
+        subel = ele.sub_elements()
+        ele = subel[0]
+
+    N = ele.degree()
+    issupported = True
+    if(type(N) == tuple):
+        issupported = len(set(N)) == 1
+        N = N[0]
+
+    if isinstance(ele, firedrake.TensorProductElement):
+        family = [K.family() for K in ele.sub_elements()]
+    else:
+        family = ele.family()
```

Comment:
When you take `set(family)` here, you'll get a set of individual characters (because strings are iterable). So prefer:

```
if isinstance(ele, firedrake.TensorProductElement):
   family = set(e.family() for e in ele.sub_elements())
else:
   family = {ele.family()}

issupported = issupported and isquad and (family <= {"Q", "DQ", "Lagrange", "Discontinuous Lagrange"})
```

Reply:
Note that "CG" and "DG" never appear at this point, because UFL normalises names.